File: exercism/javascript/grade-school/grade-school.js
const { identity, mapValues, sortBy } = require('lodash/fp')


class School {

  constructor () {
    this._roster = {}
  }

  add (name, grade) {
    this._roster[grade] = this._roster[grade] || []
    this._roster[grade].push(name)
  }

  roster () {
    return mapValues(sortBy(identity), this._roster)
  }

  grade (n) {
    return this.roster()[n] || []
  }

}

module.exports = School

File: exercism/cpp/grade-school/grade_school.cpp
#include "grade_school.h"

const std::map<int, std::vector<std::string>> grade_school::school::roster() {
  return _roster;
}

void grade_school::school::add(std::string name, int grade) {
  _roster[grade].push_back(name);
  std::sort(_roster[grade].begin(), _roster[grade].end());
}

std::vector<std::string> grade_school::school::grade(int n) {
  return _roster[n];
}
